exercises/concept/ellens-alien-game/.docs/introduction.md

```diff
@@ -0,0 +1,104 @@
+# Introduction
+
+## Classes
+
+It is time to get to one of the core paradigms of C++: object-oriented programming (OOP).
+OOP is centered around `classes` - user-defined types of data with their own set of related functions.
+We will start with the basics and will cover more advanced topics further down the syllabus tree.
+
+### Members
+
+Classes can have **member variables** and **member functions**.
+They are accessed by the **member selection** operator `.`.
+Just as variables outside of `classes`, it is advisable to initialize member variables with a value upon declaration.
+This value will then become the default for newly created objects from this class.
+
+### Encapsulation and Information Hiding
+
+Classes offer the option to restrict access to their members.
+The two basic cases are `private` and `public`.
+`private` members are not accessible from outside the class.
+`public` members can be called freely.
+All members are `private` by default and need to be explicitly marked to be usable outside of the class.
+
+### Basic example
+
+The definition of a new `class` can be seen in the following example.
+Notice the `;` after the definition.
+
+```cpp
+class Wizzard {
+  public:               // from here on all members are accessible publicly
+    inc cast_spell() {  // defines the public member function cast_spell
+      return damage;
+    }
+    std::string name{}; // defines the public member variable `name`
+  private:              // from here on all members are private
+    int damage{5};      // defines the private member variable `damage`
+};
+
+```
+
+You can access all variables from within the class.
+Take a look at `damage` inside the `cast_spell` function.
+You cannot read or change `private`  members outside of the class:
+
+```cpp
+Wizzard silverhand{};
+// calling the `cast_spell` function is okay, it is public:
+iggwilv.cast_spell();
+// => 5
+// name is public and can be changed:
+silverhand.name = "Laeral";
+// damage is private -> Compilation error
+silverhand.damage = 500; 
+```
+
+### Constructors
+
+Constructors offer the possibility to assign values to member variables at object creation.
+They have the same name as the `class` and do not have a return type.
+A class can have several constructors.
+This is useful if you do not always have a need to set all variables.
```

Comment:
That might be misleading. How about
This is useful if you want to construct the object in different ways.
But then again, that will be discussed in the constructor concept, right?

Reply:
I do plan an advanced concept for classes, probably several, with constructors, move, copy, destructor etc.

Reply:
What do you think of the addition?